addons/mail/static/src/core/common/channel_member_model.js

```diff
@@ -16,14 +16,43 @@ export class ChannelMember extends Record {
         return super.insert(...arguments);
     }
 
+    clearLocalSeenOnNextDisplay = false;
     /** @type {string} */
     create_date;
     /** @type {number} */
     id;
+    isThreadDisplayed = Record.attr(false, {
+        compute() {
+            if (this.store.discuss.isActive && !this.store.env.services.ui.isSmall) {
+                return this.thread?.eq(this.store.discuss.thread);
+            }
+            return Boolean(this.store.ChatWindow.get({ thread: this.thread }));
+        },
+        onUpdate() {
+            if (this.isThreadDisplayed && this.clearLocalSeenOnNextDisplay) {
+                this.localSeenMessage = this.seen_message_id;
+            }
+            this.clearLocalSeenOnNextDisplay = true;
+        },
+    });
     /** @type {luxon.DateTime} */
     last_interest_dt = Record.attr(undefined, { type: "datetime" });
+    localSeenMessage = Record.one("Message", {
```

Comment:
I would rename `localSeenMessage` to simply `newMessageSeparator`.

I dislike "local" vs "non-local" and keeping "seen" terminology that becomes bloated and misleading.
Naming it as `newMessageSeparator` will properly scope it to its own "new message separator" concern and remove ambiguity for everyone that see the functional feature and has to navigate on the code.

Yes, this is kept as a `Record.one("Message")`. Yes, the name doesn't have "message" word. It just adds unnecessary noise and makes naming the concept more complicated than necessary.